src/Shared/PackageManagers/NPMProjectManager.cs

```diff
@@ -131,7 +133,7 @@ public override async Task<IEnumerable<string>> EnumerateVersionsAsync(PackageUR
                 string packageName = purl.GetFullName();
                 HttpClient httpClient = CreateHttpClient();
 
-                JsonDocument doc = await GetJsonCache(httpClient, $"{ENV_NPM_API_ENDPOINT}/{packageName}", useCache);
+                JsonDocument doc = await GetJsonCache(httpClient, $"{ENV_NPM_API_ENDPOINT}/{purl.GetFullName(true)}", useCache);
```

Comment:
Nit: I personally find boolean values without names to be less convenient to understand - how about using named arguments style?

```suggestion
                JsonDocument doc = await GetJsonCache(httpClient, $"{ENV_NPM_API_ENDPOINT}/{purl.GetFullName(encoded: true)}", useCache);
```